flink-connectors/flink-connector-kafka/pom.xml

```diff
@@ -262,10 +270,116 @@ under the License.
 				<configuration>
 					<!-- Enforce single fork execution due to heavy mini cluster use in the tests -->
 					<forkCount>1</forkCount>
-					<argLine>-Xms256m -Xmx2048m -Dmvn.forkNumber=${surefire.forkNumber} -XX:-UseGCOverheadLimit</argLine>
+					<argLine>-Xms256m -Xmx2048m -Dmvn.forkNumber=${surefire.forkNumber}
+						-XX:-UseGCOverheadLimit
+					</argLine>
 				</configuration>
+				<executions>
+					<execution>
+						<id>integration-tests</id>
+						<configuration>
+							<excludes>
+								<exclude>${test.unit.pattern}</exclude>
+								<exclude>**/*E2ECase*</exclude>
+							</excludes>
+						</configuration>
+					</execution>
+				</executions>
 			</plugin>
 		</plugins>
 	</build>
 
+	<!-- Profile for running testing framework cases -->
+	<profiles>
+		<profile>
+			<id>run-testing-framework</id>
```

Comment:
Not a massive fan that this is a) optional and b) is quite a bit of boilerplate that would need to be replicated across several connectors.
Ideally, we should execute all connector tests as ITCases with minicluster in the IDE and as E2E with container on AZP. Then we would just need a way to re-execute the same tests with a different TestEnvironment. @zentol WDYT?